### PR TITLE
[TFLite] Initial support of Densify operation

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added initial support of Densify operation

### Tickets:
 - 110457

### CODE REVIEW

File: tests/layer_tests/common/tflite_layer_test_class.py [hunk 1]
```diff
@@ -30,13 +30,20 @@ def produce_tflite_model(self, framework_model, save_path):
...
+          tf.lite.OpsSet.TFLITE_BUILTINS, # enable TensorFlow Lite ops.
+          tf.lite.OpsSet.SELECT_TF_OPS # enable TensorFlow ops.
+        ]
+        tflite_model = converter.convert()
 
         tflite_model_path = os.path.join(os.path.dirname(save_path), 'model.tflite')
         with tf.io.gfile.GFile(tflite_model_path, 'wb') as f:
             f.write(tflite_model)
+        with tf.io.gfile.GFile('E:/openvino/ov_tflite/tests/layer_tests/tensorflow_lite_tests/densify.tflite', 'wb') as f:
+            f.write(tflite_model)
```
Comment (rkazants) [inline tests/layer_tests/common/tflite_layer_test_class.py:46]: please remove hard-coded path

Comment (gkrivor) [reply tests/layer_tests/common/tflite_layer_test_class.py]: It is part of a future layer test, shouldn't be in this PR. I've removed it.

File: src/frontends/tensorflow_lite/src/op/densify.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using namespace std;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow_lite {
+namespace op {
+
+OutputVector densify(const ov::frontend::tensorflow_lite::NodeContext& node) {
+    auto input = node.get_input(0);
+    return {input};
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/src/op/densify.cpp:19]: let us use existing translator `translate_identity_op` that performs identical action

File: src/frontends/tensorflow_lite/src/sparsity_info.cpp [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                             uint8_t* dest_end,
+                             const size_t row_size,
+                             const ::flatbuffers::Vector<T>* indices,
+                             const ::flatbuffers::Vector<U>* segments) {
+    U* data = reinterpret_cast<U*>(dest - row_size);  // row size will be increased at first step
+    T last_idx = ~static_cast<T>(0);
+    for (auto idx = indices->begin(); idx != indices->end(); ++idx) {
+        if (*idx <= last_idx) {
+            data = reinterpret_cast<U*>(reinterpret_cast<uint8_t*>(data) + row_size);
+            FRONT_END_GENERAL_CHECK(reinterpret_cast<uint8_t*>(data) < dest_end, "Dense data is out of bounds");
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/src/sparsity_info.cpp:25]: I think we need to replace `reinterpret_cast`, security checks will alarm us about it.

File: src/frontends/tensorflow_lite/src/sparsity_info.cpp [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                             row_size,
+                             static_cast<const ::tflite::Int32Vector*>(m_data_desc[sparse_idx].indices)->values(),
+                             static_cast<const ::tflite::Uint16Vector*>(m_data_desc[sparse_idx].segments)->values());
+            break;
+        case ::tflite::SparseIndexVector_Int32Vector:
+            read_sparse_data(m_data.data(),
+                             m_data.data() + total_size,
+                             row_size,
+                             static_cast<const ::tflite::Int32Vector*>(m_data_desc[sparse_idx].indices)->values(),
+                             static_cast<const ::tflite::Int32Vector*>(m_data_desc[sparse_idx].segments)->values());
```
Comment (rkazants) [inline src/frontends/tensorflow_lite/src/sparsity_info.cpp:151]: Am I correct that we pack sparse tensor to ov.Tensor (that is dense)? What sizes are these dense tensors and will it be efficient by memory for a given model? May be we need to propagate this sparsity below and work on sparse tensors?

File: src/frontends/tensorflow_lite/src/sparsity_info.cpp (hunk 3 above)
Comment (gkrivor) [reply src/frontends/tensorflow_lite/src/sparsity_info.cpp:25]: I'll think about it, thanks!

File: src/frontends/tensorflow_lite/src/op/densify.cpp (hunk 2 above)
Comment (gkrivor) [reply src/frontends/tensorflow_lite/src/op/densify.cpp]: removed

File: src/frontends/tensorflow_lite/src/sparsity_info.cpp (hunk 3 above)
Comment (gkrivor) [reply src/frontends/tensorflow_lite/src/sparsity_info.cpp]: Replaced by static_cast only

**Review by rkazants — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Tests failed on Windows https://github.com/openvinotoolkit/openvino/actions/runs/6960293980/job/18940653143

**Comment by gkrivor** [comment]
@ilya-lavrenov it could be due to infrastructure issues because this test requires to download a model

**Comment by ilya-lavrenov** [comment]
@gkrivor 
According to logs from CI:
```
2023-11-22T17:09:14.1829299Z [5429/9126] cmd.exe /C "cd /D C:\actions-runner\_work\openvino\openvino\openvino_build\src\frontends\tensorflow_lite\tests && C:\actions-runner\_work\_tool\Python\3.11.6\x64\python3.exe C:/actions-runner/_work/openvino/openvino/openvino/src/frontends/tensorflow_lite/tests/test_models/gen_wrapper.py C:/actions-runner/_work/openvino/openvino/openvino/src/frontends/tensorflow_lite/tests/test_models/gen_scripts/generate_pose_detector.py C:/actions-runner/_work/openvino/openvino/openvino/bin/intel64/Release/test_model_zoo/tensorflow_lite_test_models C:/actions-runner/_work/openvino/openvino/openvino/bin/intel64/Release/test_model_zoo/tensorflow_lite_test_models/generate_pose_detector_done.txt"
2023-11-22T17:09:14.1836962Z Traceback (most recent call last):
2023-11-22T17:09:14.1839533Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 1348, in do_open
2023-11-22T17:09:14.1842571Z     h.request(req.get_method(), req.selector, req.data, headers,
2023-11-22T17:09:14.1846299Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 1286, in request
2023-11-22T17:09:14.1849021Z     self._send_request(method, url, body, headers, encode_chunked)
2023-11-22T17:09:14.1853076Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 1332, in _send_request
2023-11-22T17:09:14.1855378Z     self.endheaders(body, encode_chunked=encode_chunked)
2023-11-22T17:09:14.1857537Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 1281, in endheaders
2023-11-22T17:09:14.1859018Z     self._send_output(message_body, encode_chunked=encode_chunked)
2023-11-22T17:09:14.1861036Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 1041, in _send_output
2023-11-22T17:09:14.1862326Z     self.send(msg)
2023-11-22T17:09:14.1863387Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 979, in send
2023-11-22T17:09:14.1864444Z     self.connect()
2023-11-22T17:09:14.1865553Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\http\client.py", line 1458, in connect
2023-11-22T17:09:14.1866714Z     self.sock = self._context.wrap_socket(self.sock,
2023-11-22T17:09:14.1867408Z                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1868901Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\ssl.py", line 517, in wrap_socket
2023-11-22T17:09:14.1870079Z     return self.sslsocket_class._create(
2023-11-22T17:09:14.1870720Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1872181Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\ssl.py", line 1108, in _create
2023-11-22T17:09:14.1873382Z     self.do_handshake()
2023-11-22T17:09:14.1874669Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\ssl.py", line 1379, in do_handshake
2023-11-22T17:09:14.1875814Z     self._sslobj.do_handshake()
2023-11-22T17:09:14.1877188Z ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)
2023-11-22T17:09:14.1878652Z 
2023-11-22T17:09:14.1879133Z During handling of the above exception, another exception occurred:
2023-11-22T17:09:14.1879886Z 
2023-11-22T17:09:14.1880117Z Traceback (most recent call last):
2023-11-22T17:09:14.1883258Z   File "C:\actions-runner\_work\openvino\openvino\openvino\src\frontends\tensorflow_lite\tests\test_models\gen_scripts\generate_pose_detector.py", line 21, in <module>
2023-11-22T17:09:14.1886627Z     urllib.request.urlretrieve("https://storage.googleapis.com/mediapipe-models/pose_landmarker/pose_landmarker_lite/float16/latest/pose_landmarker_lite.task", task_file_path)
2023-11-22T17:09:14.1889483Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 241, in urlretrieve
2023-11-22T17:09:14.1890873Z     with contextlib.closing(urlopen(url, data)) as fp:
2023-11-22T17:09:14.1891711Z                             ^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1893191Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 216, in urlopen
2023-11-22T17:09:14.1894504Z     return opener.open(url, data, timeout)
2023-11-22T17:09:14.1895230Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1896655Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 519, in open
2023-11-22T17:09:14.1898078Z     response = self._open(req, data)
2023-11-22T17:09:14.1898632Z                ^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1899850Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 536, in _open
2023-11-22T17:09:14.1901176Z     result = self._call_chain(self.handle_open, protocol, protocol +
2023-11-22T17:09:14.1902083Z              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1903569Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 496, in _call_chain
2023-11-22T17:09:14.1904667Z     result = func(*args)
2023-11-22T17:09:14.1905128Z              ^^^^^^^^^^^
2023-11-22T17:09:14.1906308Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 1391, in https_open
2023-11-22T17:09:14.1907641Z     return self.do_open(http.client.HTTPSConnection, req,
2023-11-22T17:09:14.1909355Z            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2023-11-22T17:09:14.1910918Z   File "C:\actions-runner\_work\_tool\Python\3.11.6\x64\Lib\urllib\request.py", line 1351, in do_open
2023-11-22T17:09:14.1912801Z 
2023-11-22T17:09:14.1913275Z     raise URLError(err)
2023-11-22T17:09:14.1913699Z 
2023-11-22T17:09:14.1914928Z urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1006)>
2023-11-22T17:09:14.1917675Z 
```
Looks like your script does not handle errors during its execution and always reports exit code 0.

**Comment by gkrivor** [comment]
@ilya-lavrenov yep, it is lightweight and doesn't expect any environment issues :) Especially according ssl certificate's problems... How frequently it fails?

**Comment by rkazants** [comment]
@dorloff, @ilya-lavrenov, do we consider to create our public storage of models? This storage will be mounted to GHA machines for tests.

**Comment by ilya-lavrenov** [comment]
> @dorloff, @ilya-lavrenov, do we consider to create our public storage of models? This storage will be mounted to GHA machines for tests.

let's formulate this task via ticket and provide usage scenario, and how will it reduce tests time, etc

**Pull request merged**